tests/director/tests/formation_api_test.go

```diff
@@ -564,6 +564,122 @@ func TestTenantFormationFlow(t *testing.T) {
 	assert.Equal(t, secondFormation, deleteUnusedFormation.Name)
 }
 
+func TestSubaccountInAtMostOneFormationOfType(t *testing.T) {
+	ctx := context.Background()
+	const (
+		firstFormation  = "FIRST"
+		secondFormation = "SECOND"
+	)
+
+	tenantId := tenant.TestTenants.GetDefaultTenantID()
+	subaccountID := tenant.TestTenants.GetIDByName(t, tenant.TestProviderSubaccount)
+
+	formationTemplateName := "create-formation-template-name"
+	formationTemplateInput := fixtures.FixFormationTemplateInput(formationTemplateName)
+
+	formationTemplateInputGQLString, err := testctx.Tc.Graphqlizer.FormationTemplateInputToGQL(formationTemplateInput)
+	require.NoError(t, err)
+
+	createFormationTemplateRequest := fixtures.FixCreateFormationTemplateRequest(formationTemplateInputGQLString)
+	output := graphql.FormationTemplate{}
+
+	t.Logf("Create formation template with name: %q", formationTemplateName)
+	err = testctx.Tc.RunOperationWithoutTenant(ctx, certSecuredGraphQLClient, createFormationTemplateRequest, &output)
+	defer fixtures.CleanupFormationTemplate(t, ctx, certSecuredGraphQLClient, output.ID)
+	require.NoError(t, err)
+	require.NotEmpty(t, output.ID)
+	require.NotEmpty(t, output.Name)
```

Comment:
As far as I understand this is not used in that constraint and it will be useful in the next constraint when we set it to global.

Can we add a check / logic that respects that in the operator now as well as it is hard to grasp the semantics without usage. Or maybe our first constraint and the operator does not need it?

Reply:
The constraint that we have implemented in this PR has `OperatorScope` tenant and the next constraint that will be implemented will have `OperatorScope` global(the checks are performed across GAs). Both have `ConstraintScope` set to FORMATION_TYPE as they apply to the formations of type EM. We need the template for mapping the constraints. 
A template wont be needed if we have constraint that applyes to all formations regardless of their type which is not the case here.